### PR TITLE
Feature/n21 76 nextcloud button

### PR DESCRIPTION
**Pull request by arnegns**
# Description

## Links to Tickets or other pull requests

https://ticketsystem.dbildungscloud.de/browse/N21-76
https://github.com/hpi-schul-cloud/dof_app_deploy/pull/252
https://ticketsystem.dbildungscloud.de/browse/N21-111

## Changes

This PR will change the default bildungscloud filesystem to nextcloud for teams, when you have the permission NEXTCLOUD_USER.
## Deployment

New envs: FEATURE_NEXTCLOUD_TEAM_FILES_ENABLED and NEXTCLOUD_REDIRECT_URL

## Screenshots of UI changes

image
image

## Approval for review

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definition of Done
More and detailed information on the *definition of done* can be found [on Confluence](https://docs.dbildungscloud.de/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

File: controllers/teams.js [hunk 1]
```diff
@@ -634,6 +635,12 @@ router.get('/:teamId', async (req, res, next) => {
 
 		const permissions = await api(req).get(`/teams/${teamId}/userPermissions/${course.user.userId}`);
 
+		const nextcloudUrl = Configuration.get('NEXTCLOUD_REDIRECT_URL') !== ''
```
Comment (mrikallab) [inline controllers/teams.js:638]: do we need a null check first?

File: views/teams/team.hbs [hunk 2]
```diff
@@ -67,6 +67,24 @@
...
+            <div class="col-sm-12">
+                <div class="pull-right">
+                    <a href="{{../filesUrl}}" class="btn btn-add btn-secondary">
+                        <i class="fa fa-folder-open" aria-hidden="true"></i>
+                        {{$t "teams._team.files.button.toTeamFiles"}}
+                    </a>
+                    <a href="{{../nextcloudUrl}}" target="_blank" class="btn btn-add btn-primary {{#unless ../nextcloudUrl}}disabled{{/unless}}">
+                        <i class="fa fa-folder-open" aria-hidden="true"></i>
+                        {{$t "teams._team.files.button.toNextcloud"}}
+                        {{!-- TODO Feature toggle for new tag --}}
```
Comment (mrikallab) [inline views/teams/team.hbs:81]: reference a ticket  to track this TODO.

File: controllers/teams.js (hunk 1 above)
Comment (arnegns) [reply controllers/teams.js:638]: no, we dont a null check. the default schema sets a default value for it

**Review by mrikallab — APPROVED** [review]

**Review by mrikallab — APPROVED** [review]

**Review by mrikallab — DISMISSED** [review]

**Review by mrikallab — COMMENTED** [review]
Please check unit tests, they failed.

File: controllers/teams.js [hunk 3]
```diff
@@ -654,10 +661,12 @@ router.get('/:teamId', async (req, res, next) => {
 				directories,
 				files,
 				filesUrl: `/files/teams/${req.params.teamId}`,
+				nextcloudUrl,
+				useNextcloud,
 				ownerId: req.params.teamId,
-				canUploadFile: true,
-				canCreateDir: true,
-				canCreateFile: true,
+				canUploadFile: !useNextcloud,
```
Comment (Metauriel) [inline controllers/teams.js:667]: is ensured that all users that were able to upload files before have access to the nextcloud? giving the permission to school teachers is not enough here.

**Review by Metauriel — CHANGES_REQUESTED** [review]

File: locales/en.json [hunk 4]
```diff
@@ -2886,7 +2886,9 @@
 					"deleteFolder": "Delete folder",
 					"renameFolder": "Rename folder",
 					"shareFolder": "Share folder",
-					"toTeamFiles": "To the team files"
+					"toTeamFiles": "To the team files",
+					"toNextcloud": "Open file area",
+					"tag": "New"
```
[the remark this answers is not in the document]
Comment (arnegns) [reply locales/en.json]: I have renamed it to newTag

File: controllers/teams.js (hunk 3 above)
Comment (arnegns) [reply controllers/teams.js:667]: We will give the nextcloud permission to all users with the role user in our next story

File: locales/de.json [hunk 5]
```diff
@@ -3162,5 +3164,10 @@
 			"theVideoConferenceHasEither": "Die Videokonferenz hat entweder noch nicht begonnen oder wurde bereits wieder beendet.",
 			"youDoNotHaveTheNecessaryAuthorization": "Dir fehlt die nötige Berechtigung, um an der Videokonferenz teilzunehmen."
 		}
+	},
+	"info": {
+		"text": {
+			"nextcloudLink": "Wir arbeiten an einem neuen Dateibereich. Dort können ab jetzt alle <b>neuen Dateien</b> gespeichert und verwaltet werden (Dateibereich öffnen). Alle Team-Mitglieder haben automatisch Zugriff auf die dort gespeicherten Dateien. Alle bereits <b>bestehenden Dateien</b> in der {{institute_title}} sind weiterhin hier verfügbar."
```
[the remark this answers is not in the document]
Comment (arnegns) [reply locales/de.json]: This was the text which we got from ux. You re right. I changed it to {{ title }}

**Review by Metauriel — APPROVED** [review]

**Pull request merged**